tests/macros/tlineinfo.nim

```diff
@@ -0,0 +1,14 @@
+# issue #5617, feature request
+# Ability to set a NimNode's lineinfo
+import macros
+
+type
+  Test = object
+
+macro mixer(n: typed): untyped =
+  let x = newIdentNode("echo")
+  x.lineInfo = n
```

Comment:
This should be ``x.lineInfo = n.lineInfo``, it doesn't make sense otherwise!

Reply:
there is no way to set `info.filename` from outside VM. We can getFile from a node but we cannot use setFile to a node.
both `macros.newNimNode` and `macros.error` use another node to obtain fileinfo.

Reply:
Then you need to rename this procedure to something else.

Reply:
I think `copyLineInfo` would be a nice name